src/core/generate.test.ts

```diff
@@ -254,15 +302,21 @@ describe("generate", () => {
       name: Name;
     }`;
 
-    const { getZodSchemasFile } = generate({
-      sourceText,
-      nameFilter: (id) => id === "Superman",
-      getSchemaName: (id) => id.toLowerCase(),
-      keepComments: true,
+    beforeAll(async () => {
+      outputFile = join(tempOpen, "superman");
+      inputPath = join(tempOpen, "superman.ts");
+      await writeFile(inputPath, sourceText);
+      const res = await generate({
+        inputPath,
+        nameFilter: (id) => id === "Superman",
+        getSchemaName: (id) => id.toLowerCase(),
+        keepComments: true,
+      });
+      getZodSchemasFile = res.getZodSchemasFile;
```

Comment:
Yeah, indeed, for unit tests like this I do prefer duplication (to a limit of course) it’s easier to read (you don’t need to jump to the `beforeAll` to understand what’s going on)

Reply:
I would abstract this part away with a util function anyway since this exact code is being used 16 times.
It's going to look something like `type ProduceTestAssets = ({ inputPath: string, outputFile: string } & GenerateProps }) =>  Promise<....>`;
I'll take a look at this today.

Reply:
I integrated a utility function to manage the creation of the files used in tests.
I also tried a version which would bundle the `generate` return but in the end this simpler solution looks more legible and maintainable.